docs/testplan/SmartSwitch-PMON-testplan.md

```diff
@@ -0,0 +1,939 @@
+
+#### Steps
+ * get_dpu_id(self):
+    * Should return ID of the DpuModule Ex: 1 on DPU0
+ * get_reboot_cause(self):
+    * Reboot the module and then execute the "show reboot-cause ..." CLIs
+    * Verify the output string shows the correct Time and Cause
+    * Limit the testing to software reboot
+ * get_state_info(self):
+    * This should return an object
+    * Stop one of the DPU containers on this DPU
+    * Execute the CLI and check the dpu-controlplane value should be down
+ * get_health_info(self):
+    * This should return an object
+    * Stop one of the DPU containers on this DPU
+    * Execute the CLI and check if the health shows the stopped container
+ 
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch:
+    get_dpu_id(self):
+    Output: When on module DPUx should return x+1
+
+    get_reboot_cause(self):
+    Output: {"Device": "DPU0", "Name": 2024_05_31_00_33_30, "Cause":  "reboot", "Time": "Fri 31 May 2024 12:29:34 AM UTC", "User": "NA", "Comment": "NA"}
+
+    get_state_info(self):
+    Output: dpu state info object
+
+    get_health_info(self):
+    Output: dpu health info object
+
+```
+#### Pass/Fail Criteria
+ * Verify that all the APIs mentioned return the expected output
+
+### 1.4 Check modified ModuleClass APIs
+
+#### Steps
+ * get_base_mac(self):
+    * Should return the base mac address of this DPU
+    * Read all DPUs mac and verify if they are unique and not None
+ * get_system_eeprom_info(self):
+    * Verify the returned dictionary key:value
+ * get_name(self):
+    * Verify if this API returns “DPU0” to “DPU7” on each of them
+ * get_description(self):
+    * Should return a string
+ * get_type(self):
+    * Should return “DPU” which is “MODULE_TYPE_DPU”
+ * get_oper_status(self):
+    * Should return the operational status of the DPU
+    * Stop one ore more containers
+    * Execute the CLI and see if it is down
+ * reboot(self, reboot_type):
+    * Issue this CLI with input “MODULE_REBOOT_DEFAULT”
+    * verify if the module reboots
+ * get_midplane_ip(self):
+    * should return the midplane IP
+
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch:
+    get_base_mac(self):
+    Output: BA:CE:AD:D0:D0:01
+
+    get_system_eeprom_info(self):
+    Output: eeprom info object
+
+    get_name(self):
+    Output: DPU2
+
+    get_description(self):
+    Output "Pensando DSC"
+
+    get_type(self):
+    Output: DPU
+
+    get_oper_status(self):
+    Output: Online
+
+    reboot(self, reboot_type):
+    Result: the DPU should reboot
+
+    get_midplane_ip(self):
+    Output: 169.254.200.1
+
+```
+#### Pass/Fail Criteria
+ *  The test result is a pass if the return value matches the expected value as shown in the "steps" and "Sample Output".
+
+### 1.5 Check SwitchModule APIs for SmartSwitch
+* This includes the appropriate new and modified ChassisClass and ModuleClass APIs 
+
+#### Steps
+ * get_dpu_id(self, name):
+    * Provide “SWITCH”
```

Comment:
Why the switch has DPU ID?

Reply:
This is no longer the case.  SWITCH is not a module any more. Fixed.